src/Umbraco.Web/PublishedCache/NuCache/ContentNode.cs

```diff
@@ -8,7 +8,7 @@ namespace Umbraco.Web.PublishedCache.NuCache
     // represents a content "node" ie a pair of draft + published versions
     // internal, never exposed, to be accessed from ContentStore (only!)
     [DebuggerDisplay("Id: {Id}, Path: {Path}")]
-    internal class ContentNode
+    public class ContentNode
```

Comment:
Does this need to be public? It's internal for a reason because its weird and I'd rather not make this into a public api

Reply:
Yes, in order to implement IDataSource, ContentNode needs to be public. I'm open to ideas on how to refactor ContentNode to make it less weird, not really sure how to improve.